app/src/main/java/com/android/unio/ui/settings/Settings.kt

```diff
@@ -52,35 +62,76 @@ fun SettingsScreen(navigationAction: NavigationAction) {
       }
 }
 
+@OptIn(ExperimentalPermissionsApi::class)
 @Composable
 fun SettingsContainer() {
   val context = LocalContext.current
+  val preferences by LocalPreferenceFlow.current.collectAsState()
+
+  /** Location Permissions * */
+  val locationPermissions =
+      rememberMultiplePermissionsState(
+          permissions =
+              listOf(
+                  Manifest.permission.ACCESS_FINE_LOCATION,
+                  Manifest.permission.ACCESS_COARSE_LOCATION))
+  val requestPermissionLauncher =
+      rememberLauncherForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) {
+          permissions ->
+        locationPermissions.launchMultiplePermissionRequest()
+      }
+
+  /** Language * */
+  val language = preferences.get<String>(AppPreferences.LANGUAGE) ?: AppPreferences.Language.default
+  val locale = Locale(language)
+  Locale.setDefault(locale)
+
+  val configuration = context.resources.configuration
+  configuration.setLocale(Locale(language))
+  configuration.setLayoutDirection(locale)
+
+  context.createConfigurationContext(configuration)
+  context.resources.updateConfiguration(configuration, context.resources.displayMetrics)
```

Comment:
Why use the deprecated method updateConfiguration?

Reply:
This is needed to support older versions of Android.